custom_components/bosch_alarm/alarm_control_panel.py

```diff
@@ -53,27 +62,30 @@ def supported_features(self) -> int:
             | AlarmControlPanelEntityFeature.ARM_AWAY
         )
 
-    async def async_alarm_disarm(self, code=None) -> None:
-        await self._panel.area_disarm(self._area_id)
-    async def async_alarm_arm_home(self, code=None) -> None:
-        await self._panel.area_arm_part(self._area_id)
-    async def async_alarm_arm_away(self, code=None) -> None:
-        await self._panel.area_arm_all(self._area_id)
+    async def async_alarm_disarm(self, code=0) -> None:
+        await self._panel.area_disarm(self._area_id, int(code))
```

Comment:
As mentioned on the other PR, since we're already storing the code in the integration configuration, pushing the checking down into the library doesn't add any security. Let's just do it directly here.